client_side/.infrastructure/infrastructure.sh

```diff
@@ -0,0 +1,351 @@
+# Prints out the treatment conditions for the experiment and optionally starts
+# training for the infrastructure and/or Tellina.
+#
+# If the experiment just started, infra_training will be started.
+# If the current treatment is "T", tellina_training will be started.
+#
+# Parameters:
+# $1: the half of the experiment to begin treatment for, can be 1 or 2.
+begin_treatment() {
+  # Sets the current treatment and task set based on the task ordering for the
+  # experiment.
+  local experiment_half=$1
+
+  if (( ${experiment_half} == 1 )); then
+    treatment="${TASK_ORDER:0:1}"
+    task_set=${TASK_ORDER:1:1}
+  else
+    treatment="${TASK_ORDER:2:1}"
+    task_set=${TASK_ORDER:3:1}
+  fi
+
+  print_treatment
+
+  if (( task_num == 0 )); then
+    if ! [[ -f "${INFRA_DIR}/.task_num" ]]; then
+      # If the user is at the very beginning and is not resuming to task 1, then
+      # enables infrastructure training.
+      INF_TRAINING=true
+
+      if [[ "$treatment" == "T" ]]; then
+        # If the treatment is T, enables Tellina training.
+        TEL_TRAINING=true
+      fi
+    fi
+  else
+    # If the treatment is T and the user is not at the beginning, enables
+    # Tellina training.
+    if [[ "$treatment" == "T" ]]; then
+      TEL_TRAINING=true
+    fi
+  fi
+}
+
+# Checks whether infrastructure training or Tellina training is enabled and
+# disables them if possible.
+#
+# Prioritizes infrastructure training.
+check_and_update_training_status() {
+  if [[ "${INF_TRAINING:-false}" == "true" ]]; then
+    if [[ "${status}" == "success" ]]; then
+      # If the user successfully did the infrastructure training, disables it.
+      unset INF_TRAINING
+
+      if [[ "${TEL_TRAINING:-false}" == "true" ]]; then
+        tellina_training
+      fi
+    else
+      # Otherwise, print the information about the training.
+      infra_training
+    fi
+  elif [[ "${TEL_TRAINING:-false}" == "true" ]]; then
+    if [[ "${status}" == "success" ]]; then
+      # if the user successfully did the Tellina training, disables it.
+      unset TEL_TRAINING
+    else
+      # Otherwise, print the information about the training.
+      tellina_training
+    fi
+  fi
+}
+
+# Trains the user on how the infrastructure itself works. This includes:
+# - User meta-commands.
+# - Tasks and diff printing.
+# - The directory that they should be performing tasks on.
+infra_training() {
+  echo "--------------------------------------------------------------------------------"
+  echo "Please follow the instructions at"
+  echo "https://homes.cs.washington.edu/~atran35/research/bash_user_experiment/server_side/training/infrastructure_training.html"
+  echo "to learn how to do the experiment."
+}
+
+# Introduces the user to Tellina and suggests a couple of known query-command
+# pairs.
+tellina_training() {
+  echo "--------------------------------------------------------------------------------"
+  echo "This half of the experiment permits you to use the Tellina tool."
```

Comment:
Why doesn't this use print_treatment instead of this line?  Either will work, I guess.

Reply:
Print treatment happens right before either `infra_training` or `tellina_training`.